File: src/test/java/org/openrewrite/python/tree/LiteralTest.java
package org.openrewrite.python.tree;

import org.intellij.lang.annotations.Language;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;
import org.openrewrite.test.RewriteTest;

import static org.openrewrite.python.Assertions.python;

public class LiteralTest implements RewriteTest {

    @ParameterizedTest
    //language=py
    @ValueSource(strings = {
      "True", "False", "42.2", "42", "\"hello world\"",
      "\"hello \\\"world\\\"\"", "'hello world'",
      "'hello \\'world\\''"
    })
    void literals(@Language("py") String lit) {
        rewriteRun(python(lit));
    }
}

File: rewrite/tests/python/literal_test.py
from rewrite.test import rewrite_run, python


# language=python
def test_none():
    rewrite_run(python("assert None"))


# language=python
def test_boolean():
    rewrite_run(python("assert True or False"))


# language=python
def test_number():
    rewrite_run(python("assert 0 or 0.0"))
